ZelBack/src/services/fluxNetworkHelper.js

```diff
@@ -1345,21 +1345,166 @@ async function purgeUFW() {
+    }
+  });
+
+  if (unrecoverable) return;
+
+  if (dockerUserChainExists) log.info('DOCKER-USER iptables chain already created');
+
+  const checkJumpToDockerChain = await cmdAsync(checkJumpChain).catch(async (checkErr) => {
+    if (checkErr.message.includes('Bad rule')) {
+      const jumpToFluxChain = 'sudo iptables -I FORWARD -j DOCKER-USER';
+      try {
+        await cmdAsync(jumpToFluxChain);
+        log.info('New FORWARDED inserted to jump to DOCKER-USER chain in iptables');
+      } catch (err) {
+        log.error('Error inserting FORWARD jump to DOCKER-USER chain');
+        // if we can't jump, we need to bail out
+        unrecoverable = true;
+      }
+    } else {
+      // if we can't check if the chain jump is ok, we should bail out
+      log.error(checkErr);
+      unrecoverable = true;
+    }
+  });
+
+  if (unrecoverable) return;
+
+  const routeCmd = 'ip -json route';
+  const rawRoutes = await cmdAsync(routeCmd).catch((err) => log.error(err));
+
+  if (!rawRoutes) return;
+
+  let routes;
   try {
-    const cmdAsync = util.promisify(nodecmd.get);
-    const dropAccessToHostNetwork = "sudo iptables -I FORWARD -i docker0 -d $(ip route | grep \"src $(ip addr show dev $(ip route | awk '/default/ {print $5}') | grep \"inet\" | awk 'NR==1{print $2}' | cut -d'/' -f 1)\" | awk '{print $1}') -j DROP";
-    await cmdAsync(dropAccessToHostNetwork).catch((error) => log.error(`Error executing dropAccessToHostNetwork command:${error}`));
-    const giveHostAccessToDockerNetwork = "sudo iptables -I FORWARD -i docker0 -d $(ip route | grep \"src $(ip addr show dev $(ip route | awk '/default/ {print $5}') | grep \"inet\" | awk 'NR==1{print $2}' | cut -d'/' -f 1)\" | awk '{print $1}') -m state --state ESTABLISHED,RELATED -j ACCEPT";
-    await cmdAsync(giveHostAccessToDockerNetwork).catch((error) => log.error(`Error executing giveHostAccessToDockerNetwork command:${error}`));
-    const giveContainerAccessToDNS = "sudo iptables -I FORWARD -i docker0 -p udp -d $(ip route | grep \"src $(ip addr show dev $(ip route | awk '/default/ {print $5}') | grep \"inet\" | awk 'NR==1{print $2}' | cut -d'/' -f 1)\" | awk '{print $1}') --dport 53 -j ACCEPT";
-    await cmdAsync(giveContainerAccessToDNS).catch((error) => log.error(`Error executing giveContainerAccessToDNS command:${error}`));
-    log.info('Access to host from containers removed');
-  } catch (error) {
-    log.error(error);
+    routes = JSON.parse(rawRoutes);
+  } catch (err) {
+    log.error('Error parsing JSON for routes... skipping');
+    return;
+  }
+
+  const defaultRoute = routes.find((route) => route.dst === 'default');
+
+  if (!defaultRoute) {
+    log.error('Unable to find default gateway... skipping');
+    return;
+  }
+
+  const localSubnet = routes.find((route) => serviceHelper.ipInSubnet(defaultRoute.gateway, route.dst));
+
+  if (!localSubnet) {
+    log.error('Unable to find local subnet... skipping');
   }
+
+  const fluxSrc = '172.23.0.0/16';
```

Comment:
each app create own network using:
const fluxNetworkOptions = {
    Name: `fluxDockerNetwork_${appname}`,
    IPAM: {
      Config: [{
        Subnet: `172.23.${number}.0/24`,
        Gateway: `172.23.${number}.1`,
      }],
    },
  };

that why i dont understood this /16

Reply:
> each app create own network using: const fluxNetworkOptions = { Name: `fluxDockerNetwork_${appname}`, IPAM: { Config: [{ Subnet: `172.23.${number}.0/24`, Gateway: `172.23.${number}.1`, }], }, };
> 
> that why i dont understood this /16

This is network subnetting. Flux docker networks allocate out of the 172.23.0.0/16 range. I.e., like you say, a Flux app docker network will have any /24 that is 172.23.x.0. These are all encapsulated by the /16.

172.23.0.0/16 is a "supernet" of the 172.23.x.0/24 networks.

What this means, is that any Flux network will be included by the 17.23.0.0/16 subnet. Which is why we can use it for the source address in the iptables rules. It would be better to do this on a per docker interface (br interface) but that is considerably more work.

It's just saying, "I want to match any address that falls in this range"